### PR TITLE
Integrate the item previewer

### PR DESCRIPTION
**Pull request by Sunbus**
Task: https://oat-sa.atlassian.net/browse/TAO-8134

Related PRs:
- https://github.com/oat-sa/extension-tao-testqti-previewer/pull/21
- https://github.com/oat-sa/extension-tao-itemqti/pull/1280

In this PR we are trying to implement new preview item component to items menu in back-office.

### CODE REVIEW

**Comment by bugalot** [comment]
Two remarks when I tested it:

1. The new item previewer requires that responses can be submitted to test scoring. In order to activate this, we have to change file /taoQtiItem/views/js/qtiCreator/plugins/menu/preview.js line 63 to 

```javascript
readOnly: false
```

However, please make sure it has no impact on the previewer in use when reviewing results in the Results extension.

2. In case we go in the authoring tool and click "preview", we can still see the old previewer in action.

File: views/js/controller/items/action.js [hunk 1]
```diff
@@ -13,22 +13,36 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2014 (original work) Open Assessment Techniologies SA
+ * Copyright (c) 2019 (original work) Open Assessment Techniologies SA
```
Comment (ferenckiss89) [inline views/js/controller/items/action.js]: 2014-2019

File: views/js/controller/routes.js [hunk 2]
```diff
@@ -13,7 +13,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2014 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
```
Comment (ferenckiss89) [inline views/js/controller/routes.js]: 2014-2019

File: views/js/controller/items/action.js (hunk 1 above)
Comment (Sunbus) [reply views/js/controller/items/action.js]: fixed

File: views/js/controller/routes.js (hunk 2 above)
Comment (Sunbus) [reply views/js/controller/routes.js]: fixed

File: views/js/controller/items/action.js [hunk 3]
```diff
@@ -13,22 +13,36 @@
...
     'uri',
     'jquery',
     'context',
-    'taoItems/preview/preview',
-    'helpers'
-], function(binder, uri, $, context, preview, helpers){
-    'use strict';
+    'helpers',
+    'taoItems/previewer/factory',
+], function(_,  module, binder, uri, $, context, helpers, previewerFactory){
```
[the remark this answers is not in the document]
Comment (Sunbus) [reply views/js/controller/items/action.js]: fixed

File: views/js/controller/preview/itemRunner.js [hunk 4]
```diff
@@ -38,33 +39,14 @@ define([
     UserInfoService,
     ItemServiceImpl,
     actionBarHook,
-    UrlParser
+    UrlParser,
+    previewerFactory,
```
[the remark this answers is not in the document]
Comment (Sunbus) [reply views/js/controller/preview/itemRunner.js]: I have no messages, but I think now should be good.

File: views/js/controller/preview/itemRunner.js [hunk 5]
```diff
@@ -114,34 +90,14 @@ define([
...
-                            //with the parent
-                            if (isCORSAllowed === true) {
-                                frame.contentWindow.__knownParent__ = true;
-                            }
-                            //then we can wait a specific event triggered from the item
-                            $(document).off('itemready').on('itemready', function () {
-                                itemApi.connect(frame);
-                                if(frame.contentWindow.__knownParent__) {
-                                    frame.contentWindow.document.body.className += ' tao-preview-scope';
+						var uri = {
```
[the remark this answers is not in the document]
Comment (Sunbus) [reply views/js/controller/preview/itemRunner.js]: this file should be deleted ( old version of previewer ), fixed.

**Comment by oatymart** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request merged**